Mathlib/Data/Matroid/Circuit.lean

```diff
@@ -319,4 +336,128 @@ lemma ext_iff_circuit {M₁ M₂ : Matroid α} :
     M₁ = M₂ ↔ M₁.E = M₂.E ∧ ∀ C, M₁.Circuit C ↔ M₂.Circuit C :=
   ⟨fun h ↦ by simp [h], fun h ↦ ext_circuit h.1 fun C hC ↦ h.2 (C := C)⟩
 
+section Elimination
+
+/-! ### Circuit Elimination -/
+
+variable {ι : Type*} (x : ι → α) {z : α} {C₁ C₂ : Set α}
```

Comment:
Can `x` be implicit?

Reply:
It can, but I think it is more commonly explicit in use cases.

Reply:
In this case, I strongly prefer it to appear with individual declarations.

Reply:
It is increasingly more common (but not 100% required) for explicit arguments to be avoided in `variable` declarations, for ease of reading. In this case, I think that rule of thumb is sensible: certainly the first point of confusion I had when reading the strong elimination lemmas was "where did this `x` thing come from?", particularly because `C0` and `C` are written there.